controllers/orders.js

```diff
@@ -0,0 +1,50 @@
+const Order = require('../models/order');
+
+module.exports = {
+    index,
+    show,
+    create,
+    delete: deleteOrder,
+};
+
+async function index(req, res) {
+    try {
+        const orders = await Order.find({}).populate('products');
+        const isEmpty = orders.length === 0;
+        res.render('orders/index', { orders, isEmpty });
+    } catch (error) {
+        res.status(500).send('Internal Server Error');
+    }
+}
+
+async function show(req, res) {
+    try {
+        const order = await Order.findById(req.params.id);
+        res.render('orders/show', { order });
+    } catch (error) {
+        res.status(500).send('Internal Server Error');
+    }
+}
+
+async function create(req, res) {
```

Comment:
see comments on route first.

you can re-write this function to:
1. get the current user(buyer) from the database
2. find the product you're trying to add from the product detail page, in the database, using `req.params.productId`
3. check if the buyer has an existing order with the seller. if yes, add item to existing order, if no then create a new order that adds the item

Reply:
function has been updated, added comments